packages/flame/lib/src/components/text_box_component.dart

```diff
@@ -241,12 +241,13 @@ class TextBoxComponent<T extends TextRenderer> extends TextComponent {
 
   Future<Image> _fullRenderAsImage(Vector2 size) {
     final recorder = PictureRecorder();
-    final c = Canvas(recorder, size.toRect());
+    final scaledSize = size * pixelRatio;
+    final c = Canvas(recorder, scaledSize.toRect());
     c.scale(pixelRatio);
     _fullRender(c);
     return recorder.endRecording().toImageSafe(
-          (width * pixelRatio).ceil(),
-          (height * pixelRatio).ceil(),
+          scaledSize.x.ceil(),
+          scaledSize.y.ceil(),
```

Comment:
this was also wrong - `width` and `height` would be referring to the previous `this.size` and not the parameter size witch is the new size. but this was not the source of the issue in my case